libcodechecker/server/permissions.py

```diff
@@ -204,8 +209,10 @@ def has_permission(self, auth_session):
         the current permission.
         """
         if not auth_session:
-            return self._permission.default_enable
-        if auth_session.is_root and self._perm_name == 'SUPERUSER':
+            # If the user does not have an auth_session it means it is a guest
+            # and the server is running in authentication disabled mode.
+            return self._permission.guest_enable
```

Comment:
Why did you create a separate variable for this functionality? Can't we return only with `True` and remove this variable.

Reply:
It is more descriptive and future-proof this way. We might end up having a permission that does **not** inherit from `SUPERUSER` and is required to have a whole other kind of authentication ruleset.

Reply:
If the server was started in unauthenticated mode, the user has all kinds of permissions. I don't think we would like to create such a permission when on a non-authenticated server the user doesn't have some permission.

Reply:
🙂 Much easier this way indeed. I've changed this.